### PR TITLE
refactor/fix: return `Invalid` mn type instead of crashing on `assert` in `GetMnType`

### PR DESCRIPTION
**Pull request by UdjinM6**
## Issue being fixed or feature implemented
We use `GetMnType` in some critical parts of code where we process data from blocks and such. Having `assert` called there makes me nervous 😅

## What was done?
We should simply consider all unknown types invalid instead of crashing here. Introduced new dummy mn type for which no real mn can ever satisfy conditions.

## How Has This Been Tested?
run tests

## Breaking Changes
n/a

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or updated relevant unit/integration/functional/e2e tests
- [ ] I have made corresponding changes to the documentation

**For repository code-owners and collaborators only**
- [ ] I have assigned this pull request to a milestone

### CODE REVIEW

File: src/evo/dmn_types.h [hunk 1]
```diff
@@ -51,8 +68,13 @@ constexpr auto HighPerformance = mntype_struct{
...
         case MnType::Regular: return dmn_types::Regular;
         case MnType::HighPerformance: return dmn_types::HighPerformance;
-        default: assert(false);
+        default: return dmn_types::Invalid;
     }
 }
 
+[[nodiscard]] constexpr const bool IsValidMnType(MnType type)
+{
+    return GetMnType(type) != dmn_types::Invalid;
```
Comment (knst) [inline src/evo/dmn_types.h:77]: I think better to iomplement it like `return (static_cast<uint16_t>(type) > MnType::LAST);`

Comment (knst) [reply src/evo/dmn_types.h:77]: And define LAST like:
...
HighPerformance = 1,
LAST = HighPerformance,
Invalid = ....

**Review by knst — CHANGES_REQUESTED** [review]

File: src/evo/dmn_types.h [hunk 2]
```diff
@@ -74,7 +64,7 @@ constexpr auto Invalid = mntype_struct{
 
 [[nodiscard]] constexpr const bool IsValidMnType(MnType type)
 {
-    return GetMnType(type) != dmn_types::Invalid;
+    return type < MnType::COUNT;
```
Comment (knst) [inline src/evo/dmn_types.h:67]: Wow! I didn't know that members of `enum class` can be compared without cast such as`static_cast<uint16_t>(type) < static_cast<uint16_t(MnType::COUNT)`
😮😮😮

**Review by knst — APPROVED** [review]
uACK

**Review by PastaPastaPasta — APPROVED** [review]
utACK for squash merge

**Review by ogabrielides — APPROVED** [review]
LGTM

**Pull request merged**